python/lsst/cp/pipe/ptc/cpExtractPtcTask.py

```diff
@@ -201,30 +201,47 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
             Output data refs to persist.
         """
         inputs = butlerQC.get(inputRefs)
+        # Ids of input list of exposures
+        inputs['inputDims'] = [expId.dataId['exposure'] for expId in inputRefs.inputExp]
+
+        # Define a 1-1 correspondence between exp.getInfo().getVisitInfo().getExposureId()
+        # and inputs['inputDims']. This will be used in `run` when saving partial datasets.
+        expIdToInputDim = {}
+        for exp, dim in zip(inputs['inputExp'], inputs['inputDims']):
+            expId = exp.getInfo().getVisitInfo().getExposureId()
```

Comment:
I think we want to get away from using `getInfo().getVisitInfo().getExposureId()`.  Is it possible to just pass the inputDims to the arrange functions, and have them be sorted as well?  This would future-proof this part of the code in case the info/VisitInfo objects change.  This would also remove the need to pass a third mapping object to the run method.

Reply:
The DataId will have the detector number and exposure ID in it already. This is not the same thing as getExposureId because getExposureId is the detector exposure ID not the exposure ID. We are actively discussing changing `getExposureId` so it would be best not to use it here.  I think for grouping you are better off creating a tuple of exposure+detector -- then you are being explicit in how you are doing your grouping using gen3 concepts and not relying on the visitInfo at all.

Reply:
If I'm not clear, I mean that you use the tuple of the exposure and detector dataId values as the key in your grouping algorithm.